### PR TITLE
lab1 added

### PR DESCRIPTION
**Pull request by serebryakov02**

### CODE REVIEW

File: lab1/Source.cpp [hunk 1]
```diff
@@ -0,0 +1,186 @@
+#include <iostream>
+#include <string>
+
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:3]: Add comments what is happening here

File: lab1/Source.cpp [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+std::string encryptCaesar(std::string&, int);
+std::string decipherCaesar(std::string&, int);
+
+std::string encryptTrithemius(std::string&);
+std::string decipherTrithemius(std::string&);
+
+int main()
+{
+    //std::string firstPhrase = "WE WORK \"HARD!\""; // MATHEMATICS IS THE GYMNASTICS OF THE MIND!
+    //std::cout << "Original phrase:\n" << firstPhrase << '\n';
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:31]: commented ? remove it

File: lab1/Source.cpp [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+    std::string thirdPhrase = "AB C!";
+    std::cout<< "Original phrase : \n" << thirdPhrase << '\n';
+    std::string encryptPhraseT = encryptTrithemius(thirdPhrase);
+    std::cout << "\nEncrypt phrase:\n" << encryptPhraseT << '\n';
+    std::cout << "\nDecipher phrase:\n" << decipherTrithemius(encryptPhraseT) << '\n';
+
+	return 0;
+}
+
+std::string encryptPolybius(std::string& originalPhrase)
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:52]: each encription - separate file(encription goes with decription in one)

Comment (ArtemLubimov) [reply lab1/Source.cpp:52]: Also it can be a separate classes

File: lab1/Source.cpp [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+{
+    bool flag;
+    std::string result;
+
+    for (int i = 0; i < originalPhrase.size(); i++)
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:57]: short comment- description of what is here

File: lab1/Source.cpp [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+                break;
+        }
+    }
+
+    return result;
+}
+
+std::string decipherPolybius(std::string& encryptedPhrase)
+{
+    int ind1, ind2;
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:82]: comments and the same comment as upper one

File: lab1/Source.cpp [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+        ind1 = encryptedPhrase[j] - '0';
+        ind2 = encryptedPhrase[j + 1] - '0';
+        result += polybiusSquare[ind1][ind2];
+    }
+
+    return result;
+}
+
+std::string encryptCaesar(std::string& originalPhrase, int shift)
+{
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:97]: comments and the same comment as upper one

File: lab1/Source.cpp [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+                break;
+            }
+        }
+    }
+
+    return result;
+}
+
+std::string decipherCaesar(std::string& encryptedPhrase, int shift)
+{
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:120]: comments and the same comment as upper one

File: lab1/Source.cpp [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+                break;
+            }
+        }
+    }
+
+    return result;
+}
+
+std::string encryptTrithemius(std::string& originalPhrase)
+{
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:143]: comments and the same comment as upper one

File: lab1/Source.cpp [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+                break;
+            }
+        }
+    }
+
+    return result;
+}
+
+std::string decipherTrithemius(std::string& encryptedPhrase)
+{
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:166]: comments and the same comment as upper one

File: lab1/Source.cpp [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+                if (temp < 0)
+                    temp = size + temp;
+                result += caesarTable[temp];
+                break;
+            }
+        }
+    }
+
+    return result;
+}
```
Comment (ArtemLubimov) [inline lab1/Source.cpp:186]: No new line in the end

File: .gitignore [hunk 11]
```diff
@@ -0,0 +1,6 @@
+x64/
+.vs/
+*.sln
+*.vcxproj
+*.vcxproj.filters
+*.vcxproj.user
```
Comment (ArtemLubimov) [inline .gitignore:6]: No new line in the end

File: lab1/Source.cpp (hunk 2 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Added comment.

File: lab1/Source.cpp (hunk 4 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Short comments added.

File: lab1/Source.cpp (hunk 5 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Short comments added.

File: lab1/Source.cpp (hunk 6 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Short comments added.

File: lab1/Source.cpp (hunk 7 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Short comments added.

File: lab1/Source.cpp (hunk 10 above)
Comment (serebryakov02) [reply lab1/Source.cpp]: Added.

File: .gitignore (hunk 11 above)
Comment (serebryakov02) [reply .gitignore]: Added.

File: lab1/src/dataProtection.cpp [hunk 12]
```diff
@@ -0,0 +1,154 @@
+#include "dataProtection.h"
```
Comment (ArtemLubimov) [inline lab1/src/dataProtection.cpp:1]: split each encryption in separate file

Comment (serebryakov02) [reply lab1/src/dataProtection.cpp]: Splited.

**Pull request merged**